test/ApplicationFactory.ts

```diff
@@ -39,6 +40,9 @@ export const createApp = (data?: Partial<ApplicationData>,
         const session: Session | undefined = data ? configureSession(createSession(cookieSecret)) : undefined;
         session?.setExtraData(APPLICATION_DATA_KEY, data);
 
+        // @ts-ignore
+        session?.data.signin_info.company_number = 'NI000000';
```

Comment:
This is needed for certain cases only, why is that added here? I am thinking having this here make is buried so deep that it will be difficult to understand how company number is set. I'd prefer to see it set explicitly for tests that need it.

Reply:
I originally wanted to add it to the `createSession` of `SessionFactory.ts`, where all other information is added however that is type checked. The current location is just after the `createSession` is called.

I cant figure out how to get access to the session through the individual tests, as the session is enclosed in the `app` that is created by this factory.

Reply:
What test needs that?

Reply:
Review Penalty Controller, the page that the middleware intercepts at the moment, if the `session?.data.signin_info.company_number` is not set, the middleware will redirect rather than let the page render